test/functional/wallet_sweep.py

```diff
@@ -0,0 +1,172 @@
+        self.assert_tx_has_output(tx_from_wallet, self.return_addr_remainder)
+
+        self.assert_balance_swept_completely(tx_from_wallet, wallet_balance_before_sweep)
+
+    def sweep_invalid_receiver_addresses(self):
+        self.log.info("Testing sweep only with specified amount")
+        self.generate_initial_utxos([12, 9])
+        assert_greater_than(self.wallet.getbalances()["mine"]["trusted"], 0)
+
+        assert_raises_rpc_error(-8, "Must provide at least one address without a specified amount" , self.wallet.sweep, [{self.return_addr_with_amount: 5}])
+
+        self.clean_up()
+
+    def sweep_invalid_amounts(self):
+        self.log.info("Try sweeping more than balance")
+        self.generate_initial_utxos([7, 14])
+        wallet_balance_before_sweep = self.wallet.getbalances()["mine"]["trusted"]
+
+        expected_tx = self.wallet.sweep(receivers=[{self.return_addr_with_amount: 5}, self.return_addr_remainder], options={"add_to_wallet": False})
+        tx = self.wallet.decoderawtransaction(expected_tx['hex'])
+        fee = 21 - sum([o["value"] for o in tx["vout"]])
+
+        assert_raises_rpc_error(-8, "Assigned more value to outputs than available funds." , self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep + 1}, self.return_addr_remainder])
+        assert_raises_rpc_error(-6, "Insufficient funds for fees after creating specified outputs.", self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep}, self.return_addr_remainder])
+        assert_raises_rpc_error(-8, "Specified output amount to {} is below dust threshold".format(self.return_addr_with_amount), self.wallet.sweep, [{self.return_addr_with_amount: 0.00000001}, self.return_addr_remainder])
+        assert_raises_rpc_error(-6, "Dynamically assigned remainder results in dust output.", self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep - fee}, self.return_addr_remainder])
+        assert_raises_rpc_error(-6, "Dynamically assigned remainder results in dust output.", self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep - fee - Decimal(0.00000010)}, self.return_addr_remainder])
+
+        self.clean_up()
+
+    def sweep_negative_effective_value(self):
+        self.nodes[0].createwallet("dustwallet")
+        dust_wallet = self.nodes[0].get_wallet_rpc("dustwallet")
+
+        self.def_wallet.sendtoaddress(dust_wallet.getnewaddress(), 0.00000400)
+        self.def_wallet.sendtoaddress(dust_wallet.getnewaddress(), 0.00000300)
+        self.generate(self.nodes[0], 1)
+        assert_greater_than(dust_wallet.getbalances()["mine"]["trusted"], 0)
+
+        assert_raises_rpc_error(-6, "Total value of UTXO pool too low to pay for sweep. Try using lower feerate or excluding uneconomic UTXOs with 'sendmax' option.", dust_wallet.sweep, receivers=[self.return_addr_remainder], fee_rate=300)
+
+        dust_wallet.unloadwallet()
+
+    def sweep_with_sendmax(self):
```

Comment:
Here also, a `self.log.info(..)` statement can be added.

Reply:
Done